lib/PreferenceList.cc

```diff
@@ -47,6 +48,40 @@ PrefListElement PreferenceList::at(SizeType index) const {
     return pref_list_.at(index);
 }
 
+void PreferenceList::set_ties(RankType rank, PreferenceList p){
+    SizeType size = p.size();
+    cur_rank_++;
+    is_tied_[rank] = true;
+    for(SizeType i=0; i<size; i++){
+        tie_map_[rank].emplace_back(rank, p.at(i).vertex); //the rank for the preference list element is index 
```

Comment:
This syntax doesn't seem correct, tie_map_[rank] is already a reference to std::vector<PrefListElement>.
We can also just say tie_map_[rank] = p, which will set the ties at index rank in tie_map_